### PR TITLE
Sort injected names in generated module code

### LINKED ISSUE #181: InflationInjectionModule should generate module entries in a stable order to prevent Gradle cache misses
I've been investigating remote cache misses in compilation tasks and saw that Gradle points to differences in generated code as the reason, `InflationInject_ViewModule.java` being one of the files in question. I ran diffs (diff file attached) comparing CI vs local output from the same SHA and noticed that the files include the same entries, but they appear in different order. The entries are located using `RoundEnvironment.getElementsAnnotatedWith`, which returns a `Set`, hence this API does not guarantee any stable ordering for the output, but should InflationInject be sorting the entries to guarantee that generated code always looks the same?

I'm seeing the same behaviour in Dagger, so curious to get some thoughts on the expected behaviour before filing an issue against Dagger.

[InflationInject_ViewModule.txt](https://github.com/cashapp/InflationInject/files/7883949/InflationInject_ViewModule.txt)

### PR DESCRIPTION
**Pull request by Egorand**
Fixes #181

### CODE REVIEW

File: inflation-inject-processor/src/test/java/app/cash/inject/inflation/processor/InflationInjectProcessorTest.kt [hunk 1]
```diff
@@ -1061,5 +1061,143 @@ class InflationInjectProcessorTest {
...
+
+    assertAbout(javaSources())
+      .that(listOf(inputViewA, inputViewB, inputModule))
+      .processedWith(InflationInjectProcessor())
+      .compilesWithoutError()
+      .and()
+      .generatesSources(expectedFactoryA, expectedFactoryB, expectedModule)
+
+    assertAbout(javaSources())
+      .that(listOf(inputViewB, inputViewA, inputModule)) // Inputs passed in reverse order.
```
Comment (JakeWharton) [inline inflation-inject-processor/src/test/java/app/cash/inject/inflation/processor/InflationInjectProcessorTest.kt:1195]: Very nice

**Review by JakeWharton — APPROVED** [review]

**Comment by JakeWharton** [comment]
Will release later. Have a few to send out.

**Comment by Egorand** [comment]
Also don't think there's any work to do for the `ViewFactory` generator, as it emits `providedKeys` in the order in which they're declared in the View constructor, which is supposed to be stable.

**Review by Rocbrad86 — COMMENTED** [review]
Nice

**Pull request merged**